karaf/init/src/main/java/org/apache/brooklyn/launcher/osgi/OsgiLauncher.java

```diff
@@ -202,8 +199,7 @@ public void setLocalBrooklynProperties(String localBrooklynProperties) {
         this.localBrooklynProperties = localBrooklynProperties;
     }
 
-    public void setDeserializingClassRenamesProvider(DeserializingClassRenamesProvider deserializingClassRenamesProvider) {
-        this.deserializingClassRenamesProvider = deserializingClassRenamesProvider;
+    public void setDeserializingClassRenamesProvider(DeserializingClassRenamesProvider.OsgiConfigLoader val) {
```

Comment:
The parameter type needs to be `DeserializingClassRenamesProvider.OsgiConfigLoader`. In blueprint.xml, the bean with id `deserializingClassRenamesProvider` has `class="org.apache.brooklyn.core.mgmt.persist.DeserializingClassRenamesProvider.OsgiConfigLoader"`, so that is the value injected for the `property name="deserializingClassRenamesProvider" ref="deserializingClassRenamesProvider"`.